include/boost/geometry/extensions/random/strategies/agnostic/uniform_linear.hpp

```diff
@@ -0,0 +1,201 @@
+// Boost.Geometry (aka GGL, Generic Geometry Library)
+
+// Copyright (c) 2019 Tinko Bartels, Berlin, Germany.
+
+// Use, modification and distribution is subject to the Boost Software License,
+// Version 1.0. (See accompanying file LICENSE_1_0.txt or copy at
+// http://www.boost.org/LICENSE_1_0.txt)
+
+#ifndef BOOST_GEOMETRY_EXTENSIONS_RANDOM_STRATEGIES_AGNOSTIC_UNIFORM_LINEAR_HPP
+#define BOOST_GEOMETRY_EXTENSIONS_RANDOM_STRATEGIES_AGNOSTIC_UNIFORM_LINEAR_HPP
+
+#include <random>
+#include <iterator>
+#include <algorithm>
+
+#include <boost/geometry/algorithms/equals.hpp>
+#include <boost/geometry/core/point_type.hpp>
+#include <boost/geometry/core/cs.hpp>
+#include <boost/geometry/algorithms/length.hpp>
+#include <boost/geometry/algorithms/line_interpolate.hpp>
+#include <boost/geometry/strategies/cartesian/line_interpolate.hpp>
+#include <boost/geometry/strategies/geographic/line_interpolate.hpp>
+#include <boost/geometry/strategies/spherical/line_interpolate.hpp>
+
+#include <boost/geometry/extensions/random/strategies/uniform_point_distribution.hpp>
+
+namespace boost { namespace geometry
+{
+
+namespace strategy { namespace uniform_point_distribution {
+
+template
+<
+    typename Point,
+    typename DomainGeometry,
+    typename InterpolationStrategy =
+        typename strategy::line_interpolate::services::default_strategy
+            <
+                typename cs_tag<DomainGeometry>::type
+            >::type
+>
+struct uniform_linear_single
```

Comment:
This also looks like an algorithm. Are there any other strategies defined for single linear geometries?